fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalOlapScanOperator.java

```diff
@@ -129,6 +132,10 @@ public List<ScalarOperator> getPrunedPartitionPredicates() {
         return prunedPartitionPredicates;
     }
 
+    public boolean isFromSplitOR() {
```

Comment:
only OlapScan is taken into consideration?  how abort Hive, Iceberg, or Hudi?

Reply:
I don't know the underlying indexing mechanism of the outer table, and the specific benefits that can be brought by doing so. The client gains a lot by splitting or predicate manually when using olap table . We can test the situation of the outer table in the furture.

Reply:
Partition pruning of external tables can benefits from this, I think FE no needs to percept the underlying indexes of external tables,  in any situation, simple predicates have more chance to pushdown to storage layer.  in fact, min-max stat in Parquet and ORC can benefit from this. so External table should be taken into consideration,too.